packages/admin/admin/src/dataGrid/excelExport/useDataGridExcelExport.tsx

```diff
@@ -0,0 +1,90 @@
+import { DocumentNode, OperationVariables, useApolloClient } from "@apollo/client";
+import { GridColDef, GridValidRowModel } from "@mui/x-data-grid";
+import { saveAs } from "file-saver";
+import * as React from "react";
+import { useIntl } from "react-intl";
+
+import { ExcelGenerationOptions, generateExcelFile } from "./generateExcelFile";
+
+export interface ExportApi {
+    loading: boolean;
+    error?: string;
+    exportGrid: () => void;
+}
+
+export type DataGridExcelExportOptions = Omit<ExcelGenerationOptions, "worksheetName"> & {
+    fileName?: string;
+    worksheetName?: string;
+};
+
+export function useDataGridExcelExport<Row extends GridValidRowModel, GQLQuery, GQLQueryVariables extends OperationVariables>(params: {
+    columns: Array<GridColDef<Row>>;
+    variables: GQLQueryVariables;
+    query: DocumentNode;
+    resolveQueryNodes: (queryData: GQLQuery) => Row[];
+    totalCount: number;
+    exportOptions?: DataGridExcelExportOptions;
+}): ExportApi {
+    const { columns, variables, query, resolveQueryNodes, totalCount, exportOptions } = params;
+    const [loading, setLoading] = React.useState(false);
+    const [error, setError] = React.useState<string | undefined>();
+    const client = useApolloClient();
+    const intl = useIntl();
+
+    const createExcelExportDownload = React.useCallback(
+        async <Row extends GridValidRowModel>(columns: Array<GridColDef<Row>>, data: Row[], exportOptions: DataGridExcelExportOptions = {}) => {
+            const {
+                fileName = intl.formatMessage({ id: "comet.dataGrid.excelExport.defaultFileName", defaultMessage: "ExcelExport" }),
+                worksheetName = intl.formatMessage({ id: "comet.dataGrid.excelExport.defaultWorksheetName", defaultMessage: "Table 1" }),
+                styling,
+            } = exportOptions;
+            const workbook = generateExcelFile<Row>(columns, data, { worksheetName, styling });
+
+            workbook.xlsx.writeBuffer().then((buffer) => {
+                saveAs(new Blob([buffer]), safeFileNameWithExtension(fileName));
+            });
+        },
+        [intl],
+    );
+
+    const exportGrid = React.useCallback(async () => {
+        setLoading(true);
+
+        try {
+            const data: Row[] = [];
+            let offset = 0;
+            do {
+                const { data: pageData } = await client.query<GQLQuery, GQLQueryVariables>({
+                    query,
+                    variables: {
+                        ...variables,
+                        offset,
+                        limit: 100,
+                    },
+                });
+                offset += 100;
```

Comment:
For added safety I think the value `100` should be a `const` used in both places here, to prevent accidentally only changing one of the values in the future.